thrift/thrift0.13/src/test/java/com/linecorp/armeria/server/thrift/ThriftCallServiceBuilderTest.java

```diff
@@ -0,0 +1,123 @@
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.linecorp.armeria.server.thrift;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.Mockito.mock;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+
+import org.junit.jupiter.api.Test;
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import testing.thrift.main.FooService.AsyncIface;
+
+/**
+ * Test for {@link ThriftCallServiceBuilder}.
+ */
+class ThriftCallServiceBuilderTest {
+    @Test
+    void nullAndEmptyCases() {
+        assertThrows(NullPointerException.class, () ->
+                ThriftCallService.builder().addService(null)
+        );
+        assertThrows(NullPointerException.class, () ->
+                ThriftCallService.builder().addService("", (Object) null)
+        );
+        assertThrows(NullPointerException.class, () ->
+                ThriftCallService.builder().addService("", null, null)
+        );
+        assertThrows(NullPointerException.class, () ->
+                ThriftCallService.builder().addService("", null, null, null)
+        );
+        assertThrows(NullPointerException.class, () ->
+                ThriftCallService.builder().addService("", (Iterable<?>) null)
+        );
+        assertThrows(IllegalArgumentException.class, () ->
+                ThriftCallService.builder().addService("", new ArrayList<>())
+        );
+    }
+
+    @Test
+    void testBuilder() {
+        final AsyncIface defaultServiceImpl = mock(AsyncIface.class);
+        final AsyncIface fooServiceImpl = mock(AsyncIface.class);
+        final AsyncIface barServiceImpl = mock(AsyncIface.class);
+        final ThriftCallService service = ThriftCallService
+                .builder()
+                .addService(defaultServiceImpl)
+                .addService("foo", fooServiceImpl)
+                .addService("foobar", fooServiceImpl)
+                .addService("foobar", barServiceImpl)
+                .addService("foobarOnce", fooServiceImpl, barServiceImpl)
+                .addService("foobarList", ImmutableList.of(fooServiceImpl, barServiceImpl))
+                .addServices(ImmutableMap.of("fooMap", fooServiceImpl, "barMap", barServiceImpl))
+                .addServices(ImmutableMap.of("fooIterableMap",
+                                             ImmutableList.of(fooServiceImpl, barServiceImpl)))
+                .build();
+        assertTrue(service.entries().containsKey(""));
+        final Iterator<?> defaultIterator = service.entries().get("").implementations.iterator();
+        assertEquals(defaultServiceImpl, defaultIterator.next());
+        assertFalse(defaultIterator.hasNext());
```

Comment:
I think the assertion logic of this test method is not easy to read. How about building a map that contains the expected keys and values and just compare it to the return value of `service.entries()`? e.g.

```java
final Map<String, List<Object>> actualEntries = service.entries().entrySet().stream()....collect(...);
final Map<String, List<Object>> expectedEntries = ImmutableMap.builder()...build();
assertThat(actualEntries).isEqualsTo(expectedEntries);
```

Reply:
I added it. 
Would you check this?